domain.yml

```diff
@@ -214,6 +214,7 @@ actions:
 - action_recipients
 - action_handoff
 - action_handoff_options
+- utter_greet
```

Comment:
```suggestion

```

Reply:
@melindaloubser1  thanks for the suggested change, but without the 'utter_greet' in the  , it crashes....
```
╭─ ~/diga/financial-demo │ on master ············································· ✔ │ took 49s │ henry Py │ system Node │ at 11:08:21 
╰─ rasa train
Traceback (most recent call last):
  File "/home/henry/.pyenv/versions/3.7.6/bin/rasa", line 8, in <module>
    sys.exit(main())
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/__main__.py", line 92, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/cli/train.py", line 76, in train
    additional_arguments=extract_additional_arguments(args),
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/train.py", line 50, in train
    additional_arguments=additional_arguments,
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/train.py", line 101, in train_async
    additional_arguments,
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/train.py", line 172, in _train_async_internal
    new_fingerprint = await model.model_fingerprint(file_importer)
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/model.py", line 290, in model_fingerprint
    domain_without_nlg = Domain.from_dict(domain_dict)
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/core/domain.py", line 162, in from_dict
    **additional_arguments,
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/core/domain.py", line 431, in __init__
    self._check_domain_sanity()
  File "/home/henry/.pyenv/versions/3.7.6/lib/python3.7/site-packages/rasa/core/domain.py", line 1049, in _check_domain_sanity
    incorrect_mappings,
rasa.core.domain.InvalidDomain: Intent 'handoff' is set to trigger action 'utter_greet', which is not defined in the domain.
```

Reply:
Hmm, I think something else is going on there though - while this might "fix" the problem you're seeing, if it's not picking up the response template from the domain we need to fix that first. I'll try it with a clean clone of the repo and see if I can duplicate, as of now I cannot.

Reply:
@melindaloubser1  thanks for the follow up. Yes, moving the utter_greet to the action list was just a glue :+1:  , and it's bypassing the intended action. Please, let me know if you need more info about my system, etc... if you can't reproduce it when cloning from scratch.

Reply:
I took a look at this further today, and it seems to be a bug in Rasa, so it should be fixed there - the issues is just during the domain sanity check, the utter_ actions do get recognized otherwise. I'll open an issue for it.

Reply:
The bug was introduced in 1.10.9, so  on 1.10.8 you could use it as is

Reply:
Issue opened here RasaHQ/rasa#6336

Reply:
> The bug was introduced in 1.10.9, so on 1.10.8 you could use it as is
may you point me how to create a test case for this so we can protect next releases from this issue ?

Reply:
you could add a test to check that a domain with a `triggers: utter_..` in it is checked correctly (the check_domain_sanity) function; probably best to move further discussion on this to the rasa repo/issue since that's where it needs to be fixed & tested